### PR TITLE
Brain imaging add legends

### PR DESCRIPTION
**Pull request by congyu-lu**
## Description
will allows users to customize color in another branch

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by airenzp** [comment]
Thanks for working on this, age overlay looks like this:
Screenshot 2024-11-12 at 2 47 14 PM

It does not seem right

File: client/plots/brainImaging.js [hunk 1]
```diff
@@ -186,9 +205,19 @@ class BrainImaging {
 			config.overlayTW?.term?.id !== previousConfig?.overlayTW?.term?.id ||
 			config.divideByTW?.term?.id !== previousConfig?.divideByTW?.term?.id
 
-		const updateL = config.settings.brainImaging.brainImageL != this.settings?.brainImageL || overlayOrDivideByChange
-		const updateF = config.settings.brainImaging.brainImageF != this.settings?.brainImageF || overlayOrDivideByChange
-		const updateT = config.settings.brainImaging.brainImageT != this.settings?.brainImageT || overlayOrDivideByChange
+		const legendFilterChange = JSON.stringify(previousConfig?.legendFilter) !== JSON.stringify(config.legendFilter)
```
Comment (airenzp) [inline client/plots/brainImaging.js:208]: I wonder why this is needed, seems that with the other variables: updateL, updateF, updateT and overlayOrDivideByChange all the changes in the state can be tracked.

**Comment by congyu-lu** [comment]
> Thanks for working on this, age overlay looks like this:
> 
> Screenshot 2024-11-12 at 2 47 14 PM
> It does not seem right

yes will need support numeric and gene term in another PR

File: client/plots/brainImaging.js (hunk 1 above)
Comment (congyu-lu) [reply client/plots/brainImaging.js:208]: legendFilterChange is for tracking the legendFilter change. e.g.Hide Male, then we need to re-render images and legends

**Review by airenzp — APPROVED** [review]
Looks good overall, thanks.

**Review by airenzp — APPROVED** [review]

**Comment by siosonel** [comment]
should remove tracking of `update*` and jsut rely on dofetch/browser caching + rx state diff logic

**Comment by airenzp** [comment]
If we do that it would rerender all the images on every request and it would compromise performance.

**Comment by siosonel** [comment]
I tested it with Congyu, the network response is cached and the image is just the usual `` tag that doesn't have to rerender. It's not readily apparent to me which rendering code is expensive, I'll need more detail to understand the performance concern.

**Comment by airenzp** [comment]
There are three images that would be rerendered, instead of one, or more if we consider divide by or overlay use cases. As of now, rendering has improved but is still slow. Yes, the cache helps if you request the same image, but it does not help otherwise. We can talk further about it. It may be easier to huddle.

**Comment by congyu-lu** [comment]
tested with Edgar and when we don't check for updateL/F/T, changing slice index for sagital plane will only trigger one request for sagital (no requests for coronal or axial). And the response size is about 100kb so not too big to cache

**Comment by airenzp** [comment]
I think I see what you mean. With the cache when you update one coordinate image even if all are requested only one request is done to the server.

**Pull request merged**